packages/jaeger-ui-components/src/TracePageHeader/TracePageHeader.test.tsx

```diff
@@ -42,23 +51,23 @@ const setup = (propOverrides) => {
 describe('TracePageHeader test', () => {
   it('should render a header ', () => {
     setup();
-
     expect(screen.getByRole('banner')).toBeInTheDocument();
   });
 
   it('should render nothing if a trace is not present', () => {
-    setup({ trace: null });
-
+    setup({ trace: null } as TracePageHeaderEmbedProps);
     expect(screen.queryByRole('banner')).not.toBeInTheDocument();
-    expect(screen.queryByRole('heading', { traceName: getTraceName(trace.spans) })).not.toBeInTheDocument();
     expect(screen.queryAllByRole('listitem')).toHaveLength(0);
     expect(screen.queryByText(/Reset Selection/)).not.toBeInTheDocument();
   });
 
   it('should render the trace title', () => {
     setup();
-
-    expect(screen.getByRole('heading', { traceName: getTraceName(trace.spans) })).toBeInTheDocument();
+    expect(
+      screen.getByRole('heading', {
+        name: (content) => content.replace(/ /g, '').startsWith(getTraceName(trace!.spans).replace(/ /g, '')),
```

Comment:
```suggestion
        name: (content) => content.startsWith(getTraceName(trace!.spans)),
```

Reply:
Without my change, sometimes this test will fail. This is because there is sometimes a discrepancy between the trace name returned by the `getTraceName` function and the trace name in the DOM, e.g.
`getTraceName` will return `serviceD: MySQL::INSERT`, but in the DOM it’s `serviceD: MySQL:: INSERT`

Reply:
ok interesting that the trace name sometimes includes a space. Not a huge issue now but may be worth checking out later